src/prefect/engine/cloud/utilities.py

```diff
@@ -16,14 +16,11 @@ def prepare_state_for_cloud(state: State) -> State:
         state._result.store_safe_value()
 
     if (
-        state.is_failed()
-        and state.cached_inputs is not None  # type: ignore
-        and all(
-            r.result_handler is not None
-            for k, r in state.cached_inputs.items()  # type: ignore
-        )
+        isinstance(state, Failed)
```

Comment:
I find this the most frustrating thing to deal with w/ mypy: our convenient state methods can't be used without an additional `assert` on the instance type.  I've tried including the assert _inside_ the method, but mypy doesn't seem to pick it up.